sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/WriteToDataSourceV2Exec.scala

```diff
@@ -631,7 +641,18 @@ private[v2] trait V2CreateTableAsSelectBaseExec extends LeafV2CommandExec {
       qe.assertCommandExecuted()
 
       table match {
-        case st: StagedTable => st.commitStagedChanges()
+        case st: StagedTable =>
+          st.commitStagedChanges()
```

Comment:
let's add a `def commitStagedTable` in `DataSourceV2Utils`, which does the commit work and also reports driver metrics. Then we can reuse this method in both `AtomicReplaceTableExec` and here.